stew/assign2.nim

```diff
@@ -0,0 +1,58 @@
+import
+  std/typetraits,
+  ./shims/macros
+
+{.push raises: [Defect].}
+
+func assignImpl[T](tgt: var openArray[T], src: openArray[T]) =
+  mixin assign
+  when supportsCopyMem(T):
+    if tgt.len > 0:
+      copyMem(addr tgt[0], unsafeAddr src[0], sizeof(tgt[0]) * tgt.len)
```

Comment:
Is it possible that the memory of `tgt` and `src` overlaps? ref https://github.com/nim-lang/Nim/pull/21661

Reply:
that's a good point, I didn't consider overlapping arrays (including self-assignment)

Reply:
https://github.com/status-im/nim-stew/pull/182